sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarOutputWriter.scala

```diff
@@ -180,3 +166,27 @@ abstract class ColumnarOutputWriter(path: String, context: TaskAttemptContext,
     outputStream.close()
   }
 }
+
+object ColumnarOutputWriter {
+  // write buffers to outputStream via tempBuffer and close buffers
+  def writeBufferedData(buffers: mutable.Queue[(HostMemoryBuffer, Long)],
+      tempBuffer: Array[Byte], outputStream: OutputStream): Unit = {
+    val toProcess = buffers.dequeueAll(_ => true)
+    try {
+      toProcess.foreach(ops => {
+        val buffer = ops._1
```

Comment:
nit:

I know this is copy and paste so you can keep it as is if you want, but...

```
toProcess.foreach { case (buffer, len) =>
  ...
```